swap/tests/testutils/bitcoind.rs

```diff
@@ -0,0 +1,137 @@
+use std::collections::HashMap;
+use testcontainers::{
+    core::{Container, Docker, Port, WaitForMessage},
+    Image,
+};
+
+#[derive(Debug)]
+pub struct Bitcoind {
+    tag: String,
+    args: BitcoindArgs,
+    ports: Option<Vec<Port>>,
+    entrypoint: Option<String>,
+    volume: String,
+}
+
+impl Image for Bitcoind {
+    type Args = BitcoindArgs;
+    type EnvVars = HashMap<String, String>;
+    type Volumes = HashMap<String, String>;
+    type EntryPoint = str;
+
+    fn descriptor(&self) -> String {
+        format!("coblox/bitcoin-core:{}", self.tag)
+    }
+
+    fn wait_until_ready<D: Docker>(&self, container: &Container<'_, D, Self>) {
+        container
+            .logs()
+            .stdout
+            .wait_for_message(&"init message: Done loading")
+            .unwrap();
+    }
+
+    fn args(&self) -> <Self as Image>::Args {
+        self.args.clone()
+    }
+
+    fn volumes(&self) -> Self::Volumes {
+        let mut volumes = HashMap::new();
+        volumes.insert(self.volume.clone(), "/home/bdk-test".to_owned());
+        volumes
+    }
+
+    fn env_vars(&self) -> Self::EnvVars {
+        HashMap::new()
+    }
+
+    fn ports(&self) -> Option<Vec<Port>> {
+        self.ports.clone()
+    }
+
+    fn with_args(self, args: <Self as Image>::Args) -> Self {
+        Bitcoind { args, ..self }
+    }
+
+    fn with_entrypoint(self, entrypoint: &Self::EntryPoint) -> Self {
+        Self {
+            entrypoint: Some(entrypoint.to_string()),
+            ..self
+        }
+    }
+
+    fn entrypoint(&self) -> Option<String> {
+        self.entrypoint.to_owned()
+    }
+}
+
+impl Default for Bitcoind {
+    fn default() -> Self {
+        Bitcoind {
+            tag: "v0.19.1".into(),
+            args: BitcoindArgs::default(),
+            ports: None,
+            entrypoint: Some("/usr/bin/bitcoind".into()),
+            volume: uuid::Uuid::new_v4().to_string(),
```

Comment:
Note: your bitcoind image will always mount a volume with this approach even if `with_volume` is not being called. 
It would be better to use `Option::None`, set it to `Some(xxx)` in `with_volume` and check in the function `volumes()` if it is set.